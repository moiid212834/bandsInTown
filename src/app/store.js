import { configureStore } from '@reduxjs/toolkit';
import themeReducer from '../features/theme/ThemeSlice';
import artistReducer from '../features/artists/ArtistSlice';
import eventsReducer from '../features/events/EventsSlice';

/**
 * Configuring the store using redux toolkit and passing reducers
 */
export const store = configureStore({
  reducer: {
    theme: themeReducer,
    artists: artistReducer,
    events: eventsReducer,
  },
});
