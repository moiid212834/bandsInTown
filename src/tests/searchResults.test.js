import React from 'react';
import {Provider} from 'react-redux';
import {store} from '../app/store';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import SearchResults from '../components/searchResults';

it('renders correctly', () => {
    const tree = renderer
        .create(
        <Provider store={store}>
            <Router>
                <SearchResults></SearchResults>
            </Router>
        </Provider>
    )
        .toJSON();
    expect(tree).toMatchSnapshot();
});