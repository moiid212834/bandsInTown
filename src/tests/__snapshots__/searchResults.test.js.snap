// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1atslg0-MuiTypography-root"
  >
    Search results for "
    "
  </p>
  <div
    className="MuiBox-root css-1yjvs5a"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-1kwywab-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-1sahu1s-MuiGrid-root"
      >
        <div
          style={
            Object {
              "display": "none",
              "marginLeft": "20px",
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": "30px",
                "fontWeight": "bolder",
              }
            }
          >
            :(
          </div>
          <h5
            style={
              Object {
                "marginTop": "5px",
              }
            }
          >
            No matching results found! Please try to search for a valid band name!
          </h5>
        </div>
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-ttpdjd-MuiPaper-root"
          style={
            Object {
              "display": "block",
            }
          }
        >
          <div
            className="css-1d9cypr-MuiStack-root"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-6rjwxe-MuiAvatar-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div>
              <h2
                className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
              />
              <div
                className="css-xr0cpo-MuiStack-root"
              >
                <button
                  className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "fontSize": "80%",
                      "fontWeight": "bold",
                      "padding": "10px",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  View Events
                </button>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge clickableIcon css-tzssek-MuiSvgIcon-root"
                  data-testid="FacebookOutlinedIcon"
                  focusable="false"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "0 20px",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
